src/logger.js

```diff
@@ -192,20 +182,27 @@ class Logger {
 
   /** @private */
   output(message, additionalArguments, outputType = loggerLevels.log) {
-    if (this.shouldSuppressOutput(outputType)) return;
-
+    if (this.shouldSuppressOutput(message, outputType)) return;
     const event = this.serializer.serialize(
-      message, additionalArguments, outputType,
+      message, additionalArguments, outputType, this.contextData,
     );
-
-    const formattedEvent = eventFormatter(event, this.logFormat);
-
-    console.log(`${this.format(formattedEvent)}`); // eslint-disable-line no-console
+    const fieldsToExpose = Object.keys(asyncLocalStorage.get('logArguments') || {})
+      .reduce((acc, key) => [...acc, { fieldName: key }], []);
+    const formattedLog = eventFormatter(event, fieldsToExpose, this.logFormat, this.logLimit);
+
+    // eslint-disable-next-line no-console
+    if (!formattedLog.chunked) console.log(`${this.format(formattedLog)}`);
+    // eslint-disable-next-line no-console
+    else formattedLog.chunks.map(chunk => console.log(`${this.format(chunk)}`));
   }
 
   /** @private */
-  shouldSuppressOutput(outputType) {
-    return [
+  shouldSuppressOutput(message, outputType) {
+    // force enable log if flipLevelPattern is matched over message
+    const flipLog = this.flipLevelPattern
```

Comment:
o `flipLog` é para forçar para não suprimir o log?

Reply:
Sim, o `flipLog` permite que, dado um pattern, o log seja emitido mesmo que o nível ou demais validações não comtemplem os dados do log. Ou seja, podemos manter o level do log em warn mas logar debug caso um determinado valor esteja presente no log, como um código de usuário específico por exemplo.